design/improve_error_handling_todo_list.md

```diff
@@ -0,0 +1,218 @@
+# Improve Error Handling - TODO List
+
+
+This doucment maps the builtin or third party exceptino that are raised and suggest which exceptions we should use instead. It is listed by files, inside of file function and inside of function an error handling flow. In addition information from this document about reasons of why we throw these exception should be part of the relevant functions in their documentation.
+
+* api.py
+
+  * init function - These two exceptions can be replaced with MissingParameterException
+
+    * EmptyParameterError Account
+
+      ```python
+      self._account = account
+      if not self._account:
+          raise RuntimeError("Account cannot be empty!")
+      ```
+
+    * EmptyParameterError Url
+
+      * ```python
+        self._account = account
+        if not self._account:
+            raise RuntimeError("Account cannot be empty!")
+        ```
+
+  * login function-
+
+    * MissingRequiredParameterException
+
+      ```python
+      if not login_id or not password:
+          # TODO: Use custom error
+          raise RuntimeError("Missing parameters in login invocation!")
+      ```
+
+  * authenticate function
+
+    * MissingRequiredParameterException
```

Comment:
Current: `RuntimeError` 
Recommended: `MissingRequiredParameterException`

is easier to understand. Please add this form throughout the doc

Reply:
Added this form:
`RuntimeError` should be replaced with `MissingParameterException`